src/core_plugins/kibana/public/discover/embeddable/search_embeddable_handler.js

```diff
@@ -0,0 +1,77 @@
+import searchTemplate from './search_template.html';
+import angular from 'angular';
+import * as columnActions from 'ui/doc_table/actions/columns';
+import { getPersistedStateId } from 'plugins/kibana/dashboard/panel/panel_state';
+import { EmbeddableHandler } from 'ui/embeddable';
+
+
+export class SearchEmbeddableHandler extends EmbeddableHandler {
+
+  constructor($compile, $rootScope, searchLoader) {
+    super();
+    this.$compile = $compile;
+    this.searchLoader = searchLoader;
+    this.$rootScope = $rootScope;
+    this.name = 'search';
+  }
+
+  async getEditPath(panelId) {
+    return await this.searchLoader.urlFor(panelId);
```

Comment:
Yeah, without a use case we probably shouldn't design a solution, I'm fine with punting for now. We can figure out how to make it work when we need it (hint: I'm thinking observables)